src/Entity/BaseGallery.php

```diff
@@ -27,6 +27,27 @@ public function __construct()
         $this->galleryHasMedias = new ArrayCollection();
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function setGalleryHasMedias($galleryHasMedias)
+    {
+        foreach ($galleryHasMedias as $galleryHasMedia) {
```

Comment:
No clear needed if we try to replease the collection?

Reply:
replace? ping @gremo you should answer this

Reply:
> replace?

AFAIK this is the default behavior in doctrine. If you want to add many childrens at once, you should create a `addAllGalleryHasMedia` method

Reply:
Yes, I was just taking a guess at what "replease" could possibly mean :P

Reply:
What I should answer?
I've just found the bug and the fix: It works with both version of Doctrine (2.5 and 2.6). But I'm waiting for some authoritative  opinion about this...

But again this is huge: think about how many people are using SonataMediaBundle with 3.4 LTS (may affect other Sonata bundles too).

Reply:
> What I should answer?

Explain to @core23 why you don't want to call `clear()` on the collection.

Reply:
Well @core23 IMHO there is no need: the collection is already up-to-date as soon as you submit the form using the admin. What clear() is supposed to do?

Reply:
> the collection is already up-to-date as soon as you submit the form using the admin. 

What if you use the `setGalleryHasMedias` outside the admin form? You could also do this by hand in a custom controller.